pantalaimon/daemon.py

```diff
@@ -578,11 +578,17 @@ async def start_pan_client(self, access_token, user, user_id, password):
             store_class=self.client_store_class,
             media_info=self.media_info,
         )
-        response = await pan_client.login(password, "pantalaimon")
 
-        if not isinstance(response, LoginResponse):
-            await pan_client.close()
-            return
+        if password is "":
+            # If password is blank, we cannot login normally and must
+            # fall back to using the provided device_id.
+            pan_client.restore_login(user_id, device_id, access_token)
```

Comment:
`device_id` can be `None` here. We should handle the `None` case either here or when getting it out of the response.

Reply:
I've added an exception, though feel free to suggest a better type / message for the exception.

Reply:
I don't think an exception is the right thing to do here since nobody will catch it up the stack either.

Returning and adding a log line here should be enough.